ci/build.sh

```diff
@@ -1,15 +1,20 @@
-#!/bin/sh -ex
+#!/bin/sh -e
 # This script is used by travis.yml and docker.sh
 
 if [ -n "${SPECIAL}" ]; then
-  sh ci/${SPECIAL}.sh
+  sh ci/"${SPECIAL}".sh
 else
-  mkdir -p ${BUILDDIR:-build}
-  cd ${BUILDDIR:-build}
-  ${CMAKE_PREFIX:-} cmake ${CCACHE_LOC:-..} ${CMAKE_PARAMS:-}
-  # 4 threads seems a reasonable default for Travis
-  ${CMAKE_PREFIX:-} cmake --build . ${BUILDEXTRAFLAGS:-} -- -j4
-  # Warning: Rare random failures when running with j4.
-  test "${RUN_TESTS:-1}" -eq "1" && ctest --output-on-failure -j1
+  JOBS=$(getconf _NPROCESSORS_ONLN 2>/dev/null) # Linux
+  [ -z "$JOBS" ] && JOBS=$(getconf NPROCESSORS_ONLN) # FreeBSD
+  [ -z "$JOBS" ] && JOBS=$(ksh93 -c 'getconf NPROCESSORS_ONLN') # Solaris
```

Comment:
Is the `nproc` command available on FreeBSD and Solaris?

Reply:
nproc is an optional package. e.g. on Alpine it's not available by default.
FreeBSD does something like `alias nproc='sysctl -n hw.ncpu'`, so no. Not by default.
Not sure about Solaris.

I found this on the internet and cannot verify FreeBSD or solaris myself. But multiple websites show this approach. e.g. https://www.mmbyte.com/article/41367.html, https://unix.stackexchange.com/questions/564494/how-to-posix-ly-find-out-how-many-cpu-threads-do-i-have-within-a-shell-script

Reply:
I feel like I'm missing a piece of the puzzle. Do you plan to add CI jobs for FreeBSD and Solaris?

Reply:
In case we pick Cirrus we might run FreeBSD jobs. But I have not found any CI that supports Solaris.

However that was not the main reason I added those lines. It was simply because it was easy to copy paste them here. They are usually mentioned together with the Linux solution when searching how to acquire the number of cores. So I did not have to research them in addition and it doesn't look like they require extra maintenance effort. So I added them.

I actually started with nproc as you suggested, but that failed on Alpine Linux without installing an additional package. I had considered that also, but it seemed overkill because this "uglier" solution is still quite trivial.

Since this script is within "ci" we can drop the other systems. I actually did not think of that. I thought people would be using this script to build, but that's wrong.